src/transformers/modeling_utils.py

```diff
@@ -2584,69 +2568,9 @@ def from_pretrained(
-                return_unused_kwargs=True,
-                **kwargs,
-            )
-        elif quantization_method_from_args == QuantizationMethod.BITS_AND_BYTES:
-            load_in_8bit = quantization_config.load_in_8bit
-            load_in_4bit = quantization_config.load_in_4bit
-
-            quantization_config_kwargs = {
-                k: v for k, v in kwargs.items() if k in inspect.signature(BitsAndBytesConfig).parameters
-            }
-
-            if len(quantization_config_kwargs) > 0:
-                raise ValueError(
-                    "You can't pass `load_in_8bit` or any other `BitsAndBytesConfig` argument as a kwarg when passing "
-                    "`quantization_config` argument at the same time."
-                )
-
-        if load_in_8bit or load_in_4bit:
-            if not (is_accelerate_available() and is_bitsandbytes_available()):
-                raise ImportError(
-                    "Using `load_in_8bit=True` requires Accelerate: `pip install accelerate` and the latest version of"
-                    " bitsandbytes `pip install -i https://test.pypi.org/simple/ bitsandbytes` or"
-                    " pip install bitsandbytes` "
-                )
-
-            if torch_dtype is None:
-                # We force the `dtype` to be float16, this is a requirement from `bitsandbytes`
-                logger.info(
-                    f"Overriding torch_dtype={torch_dtype} with `torch_dtype=torch.float16` due to "
-                    "requirements of `bitsandbytes` to enable model loading in 8-bit or 4-bit. "
-                    "Pass your own torch_dtype to specify the dtype of the remaining non-linear layers or pass"
-                    " torch_dtype=torch.float16 to remove this warning."
-                )
-                torch_dtype = torch.float16
-
-            if device_map is None:
-                if torch.cuda.is_available():
-                    device_map = {"": torch.cuda.current_device()}
-                else:
-                    raise RuntimeError("No GPU found. A GPU is needed for quantization.")
-                logger.info(
-                    "The device_map was not initialized. "
-                    "Setting device_map to {'':torch.cuda.current_device()}. "
-                    "If you want to use the model for inference, please set device_map ='auto' "
-                )
-                if low_cpu_mem_usage is None:
-                    low_cpu_mem_usage = True
-
-            if from_tf or from_flax:
-                raise ValueError(
-                    "Converting into 4-bit or 8-bit weights from tf/flax weights is currently not supported, please make"
-                    " sure the weights are in PyTorch format."
-                )
+        # processing quantization args before model_kwargs instantiated
+        quantization_config_parser = QuantizationConfigParser()
+        kwargs = quantization_config_parser.parse_config_from_args(quantization_config, **kwargs)
```

Comment:
just a heads up that this modifies the kwargs, but it only does

```python
        load_in_4bit = kwargs.pop("load_in_4bit", False)
        load_in_8bit = kwargs.pop("load_in_8bit", False)
```

So it is safe, maybe just adding a comment here would help

Reply:
added comment on this to quantizers.py::57

Reply:
OK thanks!